hail/python/test/hail/expr/test_expr.py

```diff
@@ -466,18 +466,24 @@ def test_aggregator_info_score_works_with_bgen_import(self):
 
     def test_aggregator_group_by(self):
         t = hl.Table.parallelize([
-            {"cohort": None, "pop": "EUR", "GT": hl.Call([0, 0])},
-            {"cohort": None, "pop": "ASN", "GT": hl.Call([0, 1])},
-            {"cohort": None, "pop": None, "GT": hl.Call([0, 0])},
-            {"cohort": "SIGMA", "pop": "AFR", "GT": hl.Call([0, 1])},
-            {"cohort": "SIGMA", "pop": "EUR", "GT": hl.Call([1, 1])},
-            {"cohort": "IBD", "pop": "EUR", "GT": None},
-            {"cohort": "IBD", "pop": "EUR", "GT": hl.Call([0, 0])},
-            {"cohort": "IBD", "pop": None, "GT": hl.Call([0, 1])}
-        ], hl.tstruct(cohort=hl.tstr, pop=hl.tstr, GT=hl.tcall), n_partitions=3)
+            {"cohort": None, "pop": "EUR", "GT": hl.Call([0, 0]), "x": 6},
+            {"cohort": None, "pop": "ASN", "GT": hl.Call([0, 1]), "x": 2},
+            {"cohort": None, "pop": None, "GT": hl.Call([0, 0]), "x": 4},
+            {"cohort": "SIGMA", "pop": "AFR", "GT": hl.Call([0, 1]), "x": None},
+            {"cohort": "SIGMA", "pop": "EUR", "GT": hl.Call([1, 1]), "x": 2},
+            {"cohort": "IBD", "pop": "EUR", "GT": None, "x": 0},
+            {"cohort": "IBD", "pop": "EUR", "GT": hl.Call([0, 0]), "x": 1},
+            {"cohort": "IBD", "pop": None, "GT": hl.Call([0, 1]), "x": None}
+        ], hl.tstruct(cohort=hl.tstr, pop=hl.tstr, GT=hl.tcall, x=hl.tint), n_partitions=3)
 
         r = t.aggregate(hl.struct(count=hl.agg.group_by(t.cohort, hl.agg.group_by(t.pop, hl.agg.count_where(hl.is_defined(t.GT)))),
-                                  inbreeding=hl.agg.group_by(t.cohort, hl.agg.inbreeding(t.GT, 0.1))))
+                                  inbreeding=hl.agg.group_by(t.cohort, hl.agg.inbreeding(t.GT, 0.1)),
+                                  mean=hl.agg.group_by(t.cohort, hl.agg.mean(t.x)),
+                                  mean_plus_three=hl.agg.group_by(t.cohort, hl.agg.mean(t.x) + 3),
+                                  nested_mean=hl.agg.group_by(t.cohort, hl.agg.group_by(t.pop, hl.agg.mean(t.x))),
+                                  mean_squared=hl.agg.group_by(t.cohort, hl.agg.mean(t.x) * hl.agg.mean(t.x)),
+                                  count_where=hl.agg.group_by(t.cohort, hl.agg.count_where(t.x <= 2)),
+                                  count_where_nested=hl.agg.group_by(t.cohort, hl.agg.group_by(t.pop, hl.agg.count_where(t.x <= 2)))))
```

Comment:
sorry, I know I'm being picky -- but I think we should have two different count_where conditions inside the group_by, so that expressions with different agg.filters are being tested (this was my example of something that is hard to get right).

Can we add a case for explode, too?

I can PR onto yours if you want, also

Reply:
Can you give a specific example? Also, we talked about this yesterday in check-in that there will be limitations of group_by with filter conditions and thus `count_where` and `any` because the filter occurs before the SeqOp. Therefore, groups where no values are True will not appear in the output result. We decided that this can be a known nonideal behavior for 0.2 because it will take a lot more work to fix this completely. I can add a warning to the docs.

Reply:
```
FIELD1 FIELD2
A 5
B 6
A 8
B 10
```

```
group_by(FIELD1, hl.agg.count_where(FIELD2 < 7) / agg.count_where(FIELD2 > 7))
```
should return `{'A': 1.00, 'B': 1.00}`

Reply:
Unfortunately, this is the problem I was alluding to. It returns `{'A': 1.0, 'B': None}` because the result of count_where for field2 < 7 is that 'B' isn't in the output dictionary.

Reply:
can we fix this by listing out the zero values of each aggregator and doing a `.get`?